File: collabit-client/src/shared/types/response/comment.ts
import { Comment } from "../model/Comment";
import { User } from "../model/User";

export type CommentResponse = Pick<
  Comment,
  "code" | "postCode" | "content" | "createdAt" | "parentCommentCode"
> & {
  author: Pick<User, "nickname" | "profileImage" | "githubId">;
  replies?: CommentResponse[];
};
export type CreateCommentResponse = Pick<
  Comment,
  | "code"
  | "postCode"
  | "content"
  | "createdAt"
  | "updatedAt"
  | "parentCommentCode"
>;
export type EditCommentResponse = CreateCommentResponse;

File: collabit-server/src/main/java/com/collabit/community/domain/entity/Comment.java
package com.collabit.community.domain.entity;

import com.collabit.user.domain.entity.User;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.PrePersist;
import jakarta.persistence.PreUpdate;
import java.time.LocalDateTime;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

@Entity
@Builder
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@ToString
public class Comment {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int code;

    @ManyToOne
    @JoinColumn(name = "user_code")
    private User user;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "post_code")
    private Post post;

    @Column
    private String content;

    @Column
    private LocalDateTime createdAt;

    @Column
    private LocalDateTime updatedAt;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "parent_comment_code")
    private Comment parentComment;

    @Column
    boolean isDeleted;

    @PrePersist
    public void prePersist() {
        this.createdAt = LocalDateTime.now();
        this.updatedAt = null;  // 이제 null로 유지됨
    }

    @PreUpdate
    public void preUpdate() {
        this.updatedAt = LocalDateTime.now();
    }
}
